forest_benchmarking/rpe.py

```diff
@@ -326,27 +674,91 @@ def robust_phase_estimate(xs: List, ys: List, x_stds: List, y_stds: List,
         r_std = np.sqrt(x_std ** 2 + y_std ** 2)
         if r < r_std:
             # cannot reliably place the vector an any quadrant of the circle, so terminate
+            warnings.warn("Decoherence limited estimate of phase {0:.3f} to depth {1:d}. You may "
```

Comment:
It seems important to notify the user if this condition is triggered, because it affects the bound on the error in the estimate. I'm not sure a run-time warning is the proper notification.